### PR TITLE
chore: Simplify zip file generation

### PR DESCRIPTION
**Pull request by DaveTryon**
#### Details

TL/DR: Rework how we handle packaging and signing validation of assemblies that we sign with our 3rd party certificate. 

We sign and ship 3 third party assemblies as part of the CLI:
- CommandLine.dll
- Interop.AutomationCore.dll
- Newtonsoft.Json.dll

These files are currently signed while building the `CLI` project, but not during the `CLI_Full` project, since we want to only sign them once and only have 1 version at a time. The script that generates the zip file pulls files from both the `CLI` and `CLI_Full` projects when it assembles the files, and we don't have any tool-driven validation that the zip file contains the signed versions of these assemblies. To keep the signing validation from failing on the `CLI_Full` project, we exclude these files from the list of files to validate. It feels hacky and we can do better.

With this change:
1. We still sign the files only once, in the `CLI` project
2. The `CLI_Full` project (which already builds after the `CLI` project) now includes a post-build copy of the 3 assemblies that are signed during the build of the `CLI` project
3. All assemblies of the `CLI_Full` project are validated for signing
4. The script to generate the zip file now pulls files only from the `CLI_Full` project
5. At verbose level (disabled by default), the script to generate the zip file provides a lot more clarity into the parameters as they flow through the script. It's potentially overkill but since it made it easier while working on the script, I chose to leave it in.

Log output without verbose mode (much like before, but with 1 less script input):
```
PostBuild:
  powershell -f E:\A\_work\18\s\src\..\tools\scripts\BuildZippedCLI.ps1 -SrcDir E:\A\_work\18\s\src\CLI_Full\bin\Release\netcoreapp3.1\win7-x86 -TargetDir E:\A\_work\18\s\src\CLI_Installer\bin\Release\ 
  Creating E:\A\_work\18\s\src\CLI_Installer\bin\Release\AxeWindowsCLI.zip from files in E:\A\_work\18\s\src\CLI_Full\bin\Release\netcoreapp3.1\win7-x86
  Successfully Created E:\A\_work\18\s\src\CLI_Installer\bin\Release\AxeWindowsCLI.zip
```

Log output with verbose mode enabled (this is just the output from the script, omitting the output from the cmdlet):
```
  powershell -f E:\A\_work\13\s\src\..\tools\scripts\BuildZippedCLI.ps1 -SrcDir E:\A\_work\13\s\src\CLI_Full\bin\Release\netcoreapp3.1\win7-x86 -TargetDir E:\A\_work\13\s\src\CLI_Installer\bin\Release\ 
  VERBOSE: entering Create-CLIZip
  VERBOSE:   srcDir = E:\A\_work\13\s\src\CLI_Full\bin\Release\netcoreapp3.1\win7-x86
  VERBOSE:   targetDir = E:\A\_work\13\s\src\CLI_Installer\bin\Release\
  VERBOSE:   zipFileName = AxeWindowsCLI.zip
  Creating E:\A\_work\13\s\src\CLI_Installer\bin\Release\AxeWindowsCLI.zip from files in E:\A\_work\13\s\src\CLI_Full\bin\Release\netcoreapp3.1\win7-x86
  VERBOSE:     entering Create-Zipfile
  VERBOSE:       srcDir = E:\A\_work\13\s\src\CLI_Full\bin\Release\netcoreapp3.1\win7-x86
  VERBOSE:       app = AxeWindowsCLI
  VERBOSE:       zipFile = E:\A\_work\13\s\src\CLI_Installer\bin\Release\AxeWindowsCLI.zip
  VERBOSE:       patternsToRemove = *.dev.json *.pdb
  VERBOSE:         entering Get-UniqueTempFolder
  VERBOSE:           app = AxeWindowsCLI
  VERBOSE:         exiting Get-UniqueTempFolder
  VERBOSE:       scratch = C:\Users\dlab14\AppData\Local\Temp\AxeWindowsCLI_aaf500a1-eb93-4ec5-abc9-a297b85a7985
  VERBOSE:         entering Prepare-Zipfile
  VERBOSE:           srcDir = E:\A\_work\13\s\src\CLI_Full\bin\Release\netcoreapp3.1\win7-x86
  VERBOSE:           scratch = C:\Users\dlab14\AppData\Local\Temp\AxeWindowsCLI_aaf500a1-eb93-4ec5-abc9-a297b85a7985
  VERBOSE:           patternsToRemove = *.dev.json *.pdb
  VERBOSE:             entering Add-ZipfileContents
  VERBOSE:               srcDir = E:\A\_work\13\s\src\CLI_Full\bin\Release\netcoreapp3.1\win7-x86
  VERBOSE:               scratch = C:\Users\dlab14\AppData\Local\Temp\AxeWindowsCLI_aaf500a1-eb93-4ec5-abc9-a297b85a7985
  VERBOSE:               patternToInclude = *.*
  VERBOSE:               patternsToRemove = *.dev.json *.pdb
  VERBOSE:                 entering Copy-Files
  VERBOSE:                   src = E:\A\_work\13\s\src\CLI_Full\bin\Release\netcoreapp3.1\win7-x86
  VERBOSE:                   pattern = *.*
  VERBOSE:                   dst = C:\Users\dlab14\AppData\Local\Temp\AxeWindowsCLI_aaf500a1-eb93-4ec5-abc9-a297b85a7985
  VERBOSE:                   Copying E:\A\_work\13\s\src\CLI_Full\bin\Release\netcoreapp3.1\win7-x86\*.* to C:\Users\dlab14\AppData\Local\Temp\AxeWindowsCLI_aaf500a1-eb93-4ec5-abc9-a297b85a7985
  VERBOSE:                 exiting Copy-Files
  VERBOSE:                 entering Remove-FilesByPattern
  VERBOSE:                   path = C:\Users\dlab14\AppData\Local\Temp\AxeWindowsCLI_aaf500a1-eb93-4ec5-abc9-a297b85a7985
  VERBOSE:                   pattern = *.dev.json
  VERBOSE:                 exiting Remove-FilesByPattern
  VERBOSE:                 entering Remove-FilesByPattern
  VERBOSE:                   path = C:\Users\dlab14\AppData\Local\Temp\AxeWindowsCLI_aaf500a1-eb93-4ec5-abc9-a297b85a7985
  VERBOSE:                   pattern = *.pdb
  VERBOSE:                 exiting Remove-FilesByPattern
  VERBOSE:             exiting Add-ZipfileContents
  VERBOSE:         exiting Prepare-Zipfile
  VERBOSE:         entering Create-Archive
  VERBOSE:           src = C:\Users\dlab14\AppData\Local\Temp\AxeWindowsCLI_aaf500a1-eb93-4ec5-abc9-a297b85a7985
  VERBOSE:           dst = E:\A\_work\13\s\src\CLI_Installer\bin\Release\AxeWindowsCLI.zip
  VERBOSE:         exiting Create-Archive
  VERBOSE:     exiting Create-Zipfile
  Successfully Created E:\A\_work\13\s\src\CLI_Installer\bin\Release\AxeWindowsCLI.zip
  VERBOSE: exiting Create-CLIZip
```
##### Motivation

Improve the coverage of validation tools, make build issues easier to find and fix

#### Pull request checklist

- [n/a] Addresses an existing issue: #0000

### CODE REVIEW

**Review by codeofdusk — APPROVED** [review]
Seems fine enough although I'm not super familiar with PowerShell.
